### ISSUE TITLE
GUI Stops responding 0.2.21 - postgres restart fixes it

### ISSUE THREAD

**Issue by bbrendon** [wh1te909/tacticalrmm#236]
GUI has spinning wheels and doesn't load. Restarting postgres fixes it. This has happened a few times now. I haven't figured out what triggers it.

postgres has tons of these
```
2021-01-07 13:30:42.105 PST [31297] kwjmlcmm@tacticalrmm FATAL:  remaining connection slots are reserved for non-replication superuser
connections
2021-01-07 13:30:42.105 PST [31295] kwjmlcmm@tacticalrmm FATAL:  remaining connection slots are reserved for non-replication superuser
connections
2021-01-07 13:30:42.109 PST [31294] kwjmlcmm@tacticalrmm FATAL:  remaining connection slots are reserved for non-replication superuser
connections
2021-01-07 13:30:42.118 PST [31299] kwjmlcmm@tacticalrmm FATAL:  remaining connection slots are reserved for non-replication superuser
connections
2021-01-07 13:30:42.128 PST [31301] kwjmlcmm@tacticalrmm FATAL:  remaining connection slots are reserved for non-replication superuser
connections
2021-01-07 13:30:42.134 PST [31300] kwjmlcmm@tacticalrmm FATAL:  remaining connection slots are reserved for non-replication superuser
connections
2021-01-07 13:30:42.141 PST [31303] kwjmlcmm@tacticalrmm FATAL:  remaining connection slots are reserved for non-replication superuser
connections
```

celery has some of these
```
django.db.utils.OperationalError: FATAL:  remaining connection slots are reserved for non-replication superuser connections
```

**Comment by rtwright68** [comment]
Exactly what we are seeing on our 479 agent server.  Boosted CPU and RAM.

**Comment by bradhawkins85** [comment]
Storage speed may play a part in this, not saying it is the issue but may be related.
Is the database on an SSD?

**Comment by rtwright68** [comment]
Yes, my ESXi hosts are fully SSD (HPE 380's).
Everything was working great prior to v0.2.20
Definitely noticed the spinning and the issues connecting to Agents via Mesh after that update.
Now on 0.2.21

**Comment by dinger1986** [comment]
I have noticed my host being slower since the update to 0.20 but also had issues with disk space so had put it down to that.

Space is fine now!

**Comment by bradhawkins85** [comment]
Are there any errors showing up in the browser console in dev tools?

**Comment by rtwright68** [comment]
Checked the debug log, multiple agents were reporting data overdue (across all our VPNs and locally on the same LAN):

2021-01-08 13:45:12.600 | INFO     | agents.tasks:_check_agent_service:25 - Detected crashed tacticalagent service on BCC-INK, attempting recovery
2021-01-08 13:45:12.577 | INFO     | agents.tasks:_check_agent_service:25 - Detected crashed tacticalagent service on DCB-KIOSK1, attempting recovery
2021-01-08 13:45:12.570 | INFO     | agents.tasks:_check_agent_service:25 - Detected crashed tacticalagent service on LAPTOP-RCRLC, attempting recovery
2021-01-08 13:45:12.561 | INFO     | agents.tasks:_check_agent_service:25 - Detected crashed tacticalagent service on DRK-WEB, attempting recovery
2021-01-08 13:45:12.513 | INFO     | agents.tasks:_check_agent_service:25 - Detected crashed tacticalagent service on MJ03M68T, attempting recovery
2021-01-08 13:45:12.502 | INFO     | agents.tasks:_check_agent_service:25 - Detected crashed tacticalagent service on DESKTOP-51767, attempting recovery
2021-01-08 13:45:12.496 | INFO     | agents.tasks:_check_agent_service:25 - Detected crashed tacticalagent service on EMP-SQL1, attempting recovery
2021-01-08 13:45:12.464 | INFO     | agents.tasks:_check_agent_service:25 - Detected crashed tacticalagent service on BCC-BRIX-Martin, attempting recovery
2021-01-08 13:45:12.435 | INFO     | agents.tasks:_check_agent_service:25 - Detected crashed tacticalagent service on TNP-DIGSIGN1, attempting recovery
2021-01-08 13:45:12.430 | INFO     | agents.tasks:_check_agent_service:25 - Detected crashed tacticalagent service on ASC-DGSRBV, attempting recovery
2021-01-08 13:45:12.428 | INFO     | agents.tasks:_check_agent_service:25 - Detected crashed tacticalagent service on ROC-DAD9J1, attempting recovery
2021-01-08 13:45:12.419 | INFO     | agents.tasks:_check_agent_service:25 - Detected crashed tacticalagent service on BCC-RDS2, attempting recovery
2021-01-08 13:45:12.383 | INFO     | agents.tasks:_check_agent_service:25 - Detected crashed tacticalagent service on BCI-FM1, attempting recovery

**Comment by bbrendon** [comment]
Right now I only have 20 agents and load is so low very low (0.05 to 0.10). Swap used is 0.

The server is basically fine until its suddenly not.

Unfortunately I've only used mysql in the past. Digging into postgres has proven to be quite a learning curve so far.

**Comment by rtwright68** [comment]
Our server is losing its mind today.  I ran the backup script hoping a vacuum would help.  Its essentially non-responsive on the GUI and we are receiving constant notifications of data overdue and some returning to normal.
I shut it all the way down a couple times and brought it back up but its a mess today for some reason.

**Comment by wh1te909** [comment]
please update to 0.2.22, restart VM and let me know if issue is still present

don't worry about the "detected crashed service" messages in debug log that's normal, is just a temp fix for now while im making some big changes to agent

**Comment by rtwright68** [comment]
Fantastic, thanks for the rapid fix!  My guys were all having withdrawal, lol

**Comment by rtwright68** [comment]
> please update to 0.2.22, restart VM and let me know if issue is still present
> 
> don't worry about the "detected crashed service" messages in debug log that's normal, is just a temp fix for now while im making some big changes to agent

We are good now!

**Comment by wh1te909** [comment]
@bbrendon closing as this was fixed in 0.2.22

**Comment by rtwright68** [comment]
TRMM worked great over the weekend, but went completely off the rails yesterday.  Restart did not affect it as well.  Still suspecting a memory leak?  Had to shut down the VM due to the continuous "data overdue" and then "data received" messages.  Spinning wheel and lack of agent access as well.
